Update some imports in preparation for WordPressData

mokagio opened the pull request:
Same rationale as https://github.com/wordpress-mobile/WordPress-iOS/pull/24326 and https://github.com/wordpress-mobile/WordPress-iOS/pull/24348. Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/24165, all these requirements have been discovered in https://github.com/wordpress-mobile/WordPress-iOS/pull/24378.

mokagio committed:
Remove unnecessary `#import "AccountService.h"` from `MediaService.m`

dangermattic commented:
<!--
  0 Errors
  2 Warnings: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/308">25.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Move some `MenuItem` imports from `.h` to `.m`

`MenuItem` was not used in the interface. Moving the import in the
implementation will make it cleaner to update once WordPressData is
integrated.

mokagio committed:
Remove duplicated `PostContentProvider.h` import in bridging header

mokagio committed:
Remove unused `AccountService` import from `Theme`

mokagio committed:
Remove unnecessary imports from `PostService`

Will make possible to move `PostService` to WordPressData without
carrying `CommentService`.

mokagio committed:
Update access level for methods that will soon go to WordPressData

mokagio committed:
Remove unnecessary import from `ReaderSiteService`

mokagio committed:
Remove unused import from `PostService` interface

Notice we had to update the `MenuItemAbstractPostsViewController`
implementation accordingly

mokagio committed:
Explicitly import `PostServiceOptions` and `Blog` in `PostService`

Before, they seemed to implicitly work via the bridging header import.

mokagio commented on WordPress/Classes/Models/Media.swift:
```diff
@@ -9,14 +9,14 @@ public extension Media {
     ///
     @objc
-    public func incrementAutoUploadFailureCount() {
+    func incrementAutoUploadFailureCount() {
```
If you expand the diff, you'll see the whole extension is `public`

mokagio committed:
Merge branch 'trunk' into mokagio/wordpressdata-precursor-nioeno
